### ISSUE TITLE
Tests yield OSError exceptions

### ISSUE THREAD

**Issue by ghisvail** [shoyer/h5netcdf#32]
I am getting a bunch of `OSError` exceptions when running the test suite for both the packaged version `0.3.1` and the new version `0.4.2`. Here is the full log:
```
============================= test session starts ==============================
platform linux -- Python 3.6.3rc1, pytest-3.2.1, py-1.4.34, pluggy-0.4.0
rootdir: /<<PKGBUILDDIR>>, inifile:
collected 21 items

tests/test_h5netcdf.py ..F...F..............

=================================== FAILURES ===================================
______________________ test_write_netCDF4_read_legacyapi _______________________

tmp_netcdf = '/tmp/pytest-of-root/pytest-0/test_write_netCDF4_read_legacy0/testfile.nc'

    def test_write_netCDF4_read_legacyapi(tmp_netcdf):
>       roundtrip_legacy_netcdf(tmp_netcdf, legacyapi, netCDF4)

tests/test_h5netcdf.py:343: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_h5netcdf.py:331: in roundtrip_legacy_netcdf
    read_legacy_netcdf(tmp_netcdf, read_module, write_module)
tests/test_h5netcdf.py:193: in read_legacy_netcdf
    if is_h5py_char_working(tmp_netcdf, 'z'):
tests/test_h5netcdf.py:52: in is_h5py_char_working
    assert array_equal(v, _char_array)
tests/test_h5netcdf.py:32: in array_equal
    a, b = map(np.array, (a[...], b[...]))
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
/usr/lib/python3/dist-packages/h5py/_hl/dataset.py:496: in __getitem__
    self.id.read(mspace, fspace, arr, mtype, dxpl=self._dxpl)
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
h5py/h5d.pyx:181: in h5py.h5d.DatasetID.read
    ???
h5py/_proxy.pyx:130: in h5py._proxy.dset_rw
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   OSError: Can't read data (no appropriate function for conversion path)

h5py/_proxy.pyx:84: OSError
_______________________ test_write_netCDF4_read_h5netcdf _______________________

tmp_netcdf = '/tmp/pytest-of-root/pytest-0/test_write_netCDF4_read_h5netc0/testfile.nc'

    def test_write_netCDF4_read_h5netcdf(tmp_netcdf):
        write_legacy_netcdf(tmp_netcdf, netCDF4)
>       read_h5netcdf(tmp_netcdf, netCDF4)

tests/test_h5netcdf.py:363: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_h5netcdf.py:281: in read_h5netcdf
    if is_h5py_char_working(tmp_netcdf, 'z'):
tests/test_h5netcdf.py:52: in is_h5py_char_working
    assert array_equal(v, _char_array)
tests/test_h5netcdf.py:32: in array_equal
    a, b = map(np.array, (a[...], b[...]))
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
/usr/lib/python3/dist-packages/h5py/_hl/dataset.py:496: in __getitem__
    self.id.read(mspace, fspace, arr, mtype, dxpl=self._dxpl)
h5py/_objects.pyx:54: in h5py._objects.with_phil.wrapper
    ???
h5py/_objects.pyx:55: in h5py._objects.with_phil.wrapper
    ???
h5py/h5d.pyx:181: in h5py.h5d.DatasetID.read
    ???
h5py/_proxy.pyx:130: in h5py._proxy.dset_rw
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   OSError: Can't read data (no appropriate function for conversion path)

h5py/_proxy.pyx:84: OSError
===================== 2 failed, 19 passed in 1.13 seconds ======================
```

**Comment by shoyer** [comment]
Which versions of netCDF4-Python and libnetcdf (the C library) are you using?

**Comment by ghisvail** [comment]
netcdf4-python: `1.3.0`
libnetcdf: `4.4.1.1`

**Comment by ghisvail** [comment]
Link to current bug report on the Debian BTS: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=877150

**Comment by shoyer** [comment]
This is suspiciously related to https://github.com/shoyer/h5netcdf/issues/16, but I haven't updated our test suite recently and we've been testing against libnetcdf 4.4.1.1 for a while.

So it must be a new netcdf4-python version. And indeed, using a more up-to-date netcdf4-python (1.2.9 vs 1.2.4 currently used by our test suite) reveals the same test failures: https://travis-ci.org/shoyer/h5netcdf/builds/286317833

I will investigate.

**Comment by shoyer** [comment]
Fixed in 549a4f75cadde307de22274a85e343a062e2e9e0.

This turned out to just be an overly fragile test. I've already released a new version (0.4.3) which is up on pypi.
